CCDB/include/CCDB/CcdbApi.h

```diff
@@ -139,36 +166,29 @@ class CcdbApi //: public DatabaseInterface
    * @param returnFormat The format of the returned string -> one of "text/plain (default)", "application/json", "text/xml"
    * @return The listing of folder and/or objects in the format requested
    */
-  std::string list(std::string path = "", bool latestOnly = false, std::string returnFormat = "text/plain");
+  std::string list(std::string const& path = "", bool latestOnly = false, std::string const& returnFormat = "text/plain") const;
+
+  /**
+   * Make a local snapshot of all valid objects, given a timestamp, of the CCDB under a given local path.
+   * This is doing a recursive list and fetching the files locally.
+   */
+  void snapshot(std::string const& ccdbrootpath, std::string const& localDir, long timestamp) const;
 
   /**
    * Check whether the url is reachable.
    * @param url The url to test.
    * @return a bool indicating whether the url is reachable or not.
    */
-  bool isHostReachable();
+  bool isHostReachable() const;
 
  private:
-  /**
-   * Get the current timestamp.
-   *
-   * @return the current timestamp as a long
-   */
-  long getCurrentTimestamp();
   /**
    * Transform and return a string representation of the given timestamp.
    *
    * @param timestamp
    * @return a string representation of the given timestamp.
    */
-  std::string getTimestampString(long timestamp);
-  /**
-   * Compute and return a timestamp X seconds in the future.
-   *
-   * @param secondsInFuture The number of seconds in the future.
-   * @return the future timestamp
-   */
-  long getFutureTimestamp(int secondsInFuture);
+  std::string getTimestampString(long timestamp) const;
```

Comment:
Maybe this could go to CCDBTimeStampUtils.h as well

Reply:
This one seems to me as something only used in the implementation inside CcdbApi (and is private). Would leave it there for the moment

Reply:
I agree that it is only used here. I was proposing to move it in order to clarify the responsibilities of the classes.